Source/DafnyLanguageServer/Util/PathExtensions.cs

```diff
@@ -62,24 +52,24 @@ public static class PathExtensions {
     /// <param name="token">The token to get the boogie token from.</param>
     /// <returns>The uri of the document where the token is located.</returns>
     public static DocumentUri GetDocumentUri(this Boogie.IToken token) {
-      if (token is IncludeToken includeToken) {
-        return DocumentUri.FromFileSystemPath(includeToken.Include.CanonicalPath);
-      }
-
       while (token is RefinementToken refinementToken) {
         token = refinementToken.WrappedToken;
       }
 
-      return DocumentUri.Parse(token.filename);
+      return DocumentUri.From(((IToken)token).Uri);
     }
 
     /// <summary>
     /// Gets the filename of the document where the given boogie token is located in.
     /// </summary>
     /// <param name="token">The token to get the document filename from.</param>
     /// <returns>The filename (without path) of the document containing the given token.</returns>
-    public static string GetDocumentFileName(this Boogie.IToken token) {
-      return Path.GetFileName(token.filename);
+    public static string? GetDocumentFileName(this Boogie.IToken token) {
+      if (token.filename == null) {
```

Comment:
Same here, I would like the token's type to be filtered to ensure the cast does not throw a cast exception